File: Ahorn/effects/FemtoStars.jl
module FemtoHelperFemtoStars

using ..Ahorn, Maple

@mapdef Effect "FemtoHelper/FemtoStars" FemtoStars(only::String="*", exclude::String="", trailCount::Integer=8, colors::String="008080", minXSpeed::Number=0.0, maxXSpeed::Number=0.0, minYSpeed::Number=12.0, maxYSpeed::Number=24.0, extraLoopBorderX::Number=0.0, extraLoopBorderY::Number=0.0, starCount::Integer=100, backgroundColor::String="000000", backgroundAlpha::Number=1.0, sprite::String="bgs/02/stars", scrollX::Number=0.0, scrollY::Number=0.0, trailSeparation::Number=1.0, animationRate::Number=2.0, alphas::String="0.7")

placements = FemtoStars

function Ahorn.canFgBg(FemtoStars)
    return true, true
end

end

File: Loenn/effects/femto_stars.lua
local femto_stars = {}

femto_stars.name = "FemtoHelper/FemtoStars"
femto_stars.fieldInformation = {
    backgroundColor = {
        fieldType = "color",
    },
    starCount = {
        fieldType = "integer"
    },
    trailCount = {
        fieldType = "integer"
    }
}

femto_stars.defaultData = {
    trailCount = 8,
    colors = "008080FF",
    minXSpeed = 0.0,
    maxXSpeed = 0.0,
    minYSpeed = 12.0,
    maxYSpeed = 24.0,
    extraLoopBorderX = 0.0,
    extraLoopBorderY = 0.0,
    starCount = 100,
    backgroundColor = "000000",
    backgroundAlpha = 1.0,
    sprite = "bgs/02/stars",
    scrollX = 0.0,
    scrollY = 0.0,
    trailSeparation = 1.0,
    animationRate = 2.0,
    alphas = "0.7"
}

return femto_stars